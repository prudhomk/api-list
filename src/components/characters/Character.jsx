import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, species, phrase, skill, personality, image }) => (
  <figure>
    <img src={image} alt={name} />
    <figcaption>
      <p>{name}</p>
      <p>{species}</p>
      <p>{phrase}</p>
      <p>{skill}</p>
      <p>{personality}</p>
    </figcaption>
  </figure>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  phrase: PropTypes.string.isRequired,
  skill: PropTypes.string.isRequired,
  personality: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Character;
