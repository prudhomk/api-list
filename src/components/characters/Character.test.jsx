import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
    it('renders Character', () => {
        const { asFragment } = render(
            <Character
            name="Booger"
            species="Snot"
            phrase="Ooga Booga"
            skill="Picking Nose"
            personality="Snotty"
            image="https://image.com/image.jpeg"
            />
        );
    });
});