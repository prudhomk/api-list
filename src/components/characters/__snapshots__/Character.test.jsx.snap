// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Booger"
      src="https://image.com/image.jpeg"
    />
    <figcaption>
      <p>
        Booger
      </p>
      <p>
        Snot
      </p>
      <p>
        Ooga Booga
      </p>
      <p>
        Picking Nose
      </p>
      <p>
        Snotty
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
